src/v/v8_engine/instance.cc

```diff
@@ -0,0 +1,139 @@
+                                                _isolate,
+                                                script.begin(),
+                                                v8::NewStringType::kNormal,
+                                                script.size())
+                                                .ToLocalChecked();
+          v8::Local<v8::Script> compiled_script;
+          if (!v8::Script::Compile(local_ctx, script_code)
+                 .ToLocal(&compiled_script)) {
+              v8::String::Utf8Value error(_isolate, try_catch.Exception());
+              return ss::make_exception_future(exception(fmt::format(
+                "Can not compile script:{}",
+                std::string(*error, error.length()))));
+          }
+
+          v8::Local<v8::Value> result;
+          if (!compiled_script->Run(local_ctx).ToLocal(&result)) {
+              return ss::make_exception_future(
+                exception(fmt::format("Can not run script in first time")));
+          }
+
+          _context.Reset(_isolate, local_ctx);
+          return ss::now();
+      })
+      .handle_exception_type([](std::filesystem::filesystem_error& e) {
+          return ss::make_exception_future(
+            exception(fmt::format("Error: {}", e.what())));
+      });
+}
+
+ss::future<> instance::get_function(std::string_view name) {
+    v8::Locker locker(_isolate);
+    v8::Isolate::Scope isolate_scope(_isolate);
+    v8::HandleScope handle_scope(_isolate);
+    v8::Local<v8::Context> local_ctx = v8::Local<v8::Context>::New(
+      _isolate, _context);
+    v8::Context::Scope context_scope(local_ctx);
+    v8::Local<v8::String> function_name
+      = v8::String::NewFromUtf8(_isolate, name.data()).ToLocalChecked();
+    v8::Local<v8::Value> function_val;
+    if (
+      !local_ctx->Global()->Get(local_ctx, function_name).ToLocal(&function_val)
+      || !function_val->IsFunction()) {
+        return ss::make_exception_future(
+          exception(fmt::format("Can not get function from script")));
+    }
+
+    _function.Reset(_isolate, function_val.As<v8::Function>());
+    return ss::now();
+}
+
+// It is the first version for run. In next updates I use alient_thread for run
+// v8.
+ss::future<> instance::run(ss::temporary_buffer<char>& data) {
+    return run_internal(data);
+}
+
+ss::future<> instance::run_internal(ss::temporary_buffer<char>& data) {
+    v8::Locker locker(_isolate);
+    v8::Isolate::Scope isolate_scope(_isolate);
+    v8::HandleScope handle_scope(_isolate);
+    v8::TryCatch try_catch(_isolate);
+    v8::Local<v8::Context> local_ctx = v8::Local<v8::Context>::New(
+      _isolate, _context);
+    v8::Context::Scope context_scope(local_ctx);
+
+    const int argc = 1;
+    auto store = v8::ArrayBuffer::NewBackingStore(
+      data.get_write(), data.size(), v8::BackingStore::EmptyDeleter, nullptr);
+    auto data_array_buf = v8::ArrayBuffer::New(_isolate, std::move(store));
```

Comment:
Do we write anything into the ArrayBuffer from the script?

Reply:
what about the size of the buffer then? do we know how big the temporary_buffer should be?

Reply:
The caller know it.